tests/commons/j2/test_j2_functions.py

```
@@ -272,6 +275,101 @@
+    ],
+)
+def test_auto_number(case: tuple[str, str, str | None]) -> None:
+    assertCase("auto-number", case, [])  # no prefix-variants to test
+
+
+def test_fail_auto_number() -> None:
+    with pytest.raises(ValueError):
+        xsd_fmt("3.14.15", "auto-number")
+        raise Exception(
+            "auto-number should not accept strings that have no meaningfull date format"
+        )
```

Ruff (0.8.2)

`pytest.raises()` block should contain a single simple statement

(PT012)